hoomd/hpmc/pytest/test_convex_polygon.py

```diff
@@ -1,25 +1,132 @@
-    test_convex_polygon3 = hpmc.poly2d_verts(args_3)
+
+    args_3 = {'vertices': [(0, 0), (1, 1), (1, 0), (0, 1),
+                           (1, 1), (0, 0), (2, 1), (1, 3)],
+              'ignore_statistics': 1,
+              'sweep_radius': 1.5}
+    test_convex_polygon3 = _hpmc.PolygonVertices(args_3)
     test_dict3 = test_convex_polygon3.asDict()
     assert test_dict3 == args_3
-    
-    args_4 = {'vertices':[(0, 0), (1, 1), (1, 0), (0, 1), (1, 1), (0, 0), (2, 1), (1, 3), (9, 8), (1, 1)], 'ignore_statistics':0}
-    test_convex_polygon4 = hpmc.poly2d_verts(args_4)
+
+    args_4 = {'vertices': [(0, 0), (1, 1), (1, 0), (0, 1), (1, 1),
+                           (0, 0), (2, 1), (1, 3), (9, 8), (1, 1)],
+              'ignore_statistics': 0,
+              'sweep_radius': 2.0}
+    test_convex_polygon4 = _hpmc.PolygonVertices(args_4)
     test_dict4 = test_convex_polygon4.asDict()
     assert test_dict4 == args_4
+
+
+# these tests are for the python side
+def test_convex_polygon_python():
+
+    verts = [(-1, 1), (1, -1), (1, 1), (-1, -1)]
+    poly = hpmc.integrate.ConvexPolygon(23456)
+    poly.shape['A'] = dict(vertices=verts)
+    assert not poly.shape['A']['ignore_statistics']
+    np.testing.assert_allclose(poly.shape['A']['vertices'], verts)
+
+
+def test_convex_poly_after_attaching(device, dummy_simulation_factory):
+
+    verts = [(-1, 1), (1, -1), (1, 1), (-1, -1)]
+    verts2 = [(-1, 1), (1, -1), (1, 1)]
+    poly = hpmc.integrate.ConvexPolygon(23456)
+    poly.shape['A'] = dict(vertices=verts)
+    poly.shape['B'] = dict(vertices=verts2, ignore_statistics=True)
+
+    sim = dummy_simulation_factory(particle_types=['A', 'B'])
+    sim.operations.add(poly)
+    sim.operations.schedule()
+
+    assert not poly.shape['A']['ignore_statistics']
+    assert poly.shape['B']['ignore_statistics']
+    np.testing.assert_allclose(poly.shape['A']['vertices'], verts)
+    np.testing.assert_allclose(poly.shape['B']['vertices'], verts2)
+
+
+def test_overlaps(device, dummy_simulation_check_overlaps):
+    mc = hoomd.hpmc.integrate.ConvexPolygon(23456, d=0, a=0)
+    mc.shape['A'] = dict(vertices=[(0, (0.75**0.5) / 2),
+                                   (-0.5, -(0.75**0.5) / 2),
+                                   (0.5, -(0.75**0.5) / 2)])
+    sim = dummy_simulation_check_overlaps()
+    sim.operations.add(mc)
+    # gsd_dumper = hoomd.dump.GSD(filename='/Users/danevans/hoomd/
+    # test_dump_convex_polygon.gsd', trigger=1, overwrite=True)
+    # gsd_logger = hoomd.logger.Logger()
+    # gsd_logger += mc
+    # gsd_dumper.log = gsd_logger
+    # sim.operations.add(gsd_dumper)
+    sim.operations.schedule()
+    sim.run(1)
+    overlaps = sim.operations.integrator.overlaps
+    assert overlaps > 0
+
+    s = sim.state.snapshot
+    s.particles.position[0] = (0, 0, 0)
+    s.particles.position[1] = (0, 8, 0)
+    sim.state.snapshot = s
+    sim.operations.add(mc)
```

Comment:
There is no need to re-add and re-schedule the simulation after changing the snapshot.